src/main.cpp

```diff
@@ -1756,7 +1756,12 @@ bool ConnectBlock(const CBlock& block, CValidationState& state, CBlockIndex* pin
         return true;
     }
 
-    bool fScriptChecks = pindex->nHeight >= Checkpoints::GetTotalBlocksEstimate();
+    bool fScriptChecks = true;
+    CBlockIndex *pindexLastCheckpoint = Checkpoints::GetLastCheckpoint();
+    if (pindexLastCheckpoint && pindex->GetAncestor(pindexLastCheckpoint->nHeight) == pindexLastCheckpoint) {
+        // This block is an ancestor of a checkpoint: disable script checks
+        fScriptChecks = false;
+    }
```

Comment:
This code section confuses me. Appears that the current block being evaluated isn't even referenced in the code? The functions must be dependent on which block is being connected.